napari/_qt/containers/qt_layer_model.py

```diff
@@ -15,6 +18,13 @@ def data(self, index: QModelIndex, role: Qt.ItemDataRole):
         if not index.isValid():
             return None
         layer = self.getItem(index)
+        viewer = current_viewer()
+        layer_loaded = layer.loaded
+        if viewer:
+            viewer_playing = viewer.window._qt_viewer.dims.is_playing
+            force_sync = viewer._layer_slicer._force_sync
+            if not force_sync:
+                layer_loaded = layer.loaded and not viewer_playing
```

Comment:
Prefer to use the public setting instead of the private state for checking is async slicing is enabled. I think the following is equivalent and slightly simpler:

```python
        # Playback with async slicing causes flickering between the thumbnail
        # and loading animation in some cases due quick changes in the loaded state,
        # so report as unloaded in that case to avoid that.
        if get_settings().experimental.async_ and (viewer := current_viewer()):
            viewer_playing = viewer.window._qt_viewer.dims.is_playing
            layer_loaded = layer.loaded and not viewer_playing
```

This might be nit-picky, but I would consider moving this special case logic for `loaded` to the `LayerDelegate` because this is a specific workaround for a rendering bug. I feel like the model should just report the underlying value. Note that, I can't reproduce similar flickering for the tooltip, so I think that could just use `Layer.loaded` directly in that case.

Feel free to push back here!

Reply:
Will do the code change, indeed simplier! :+1: Regarding the code placement, I left this logic in the Qt model since I think the delegate should just query the model data, paint it and enable editing it (initial though that made me rethink all the timer logic).

In fact, I was thinking that it could be nice to have some sort of `layer.is_being_processed` attribute that enables to know not only if the layer is loaded but also if some continue action is being done over it, like the animation/successive slicing requests (maybe extending the loaded attribute with logic that can handle such cases could be worthy?).

Regarding the tooltip, I also used the workaround to take into account the playback state since you can end up seeing the `(loading)` text appear and disappear if you move the mouse over the layer item in the list:

![load_tooltip](https://github.com/napari/napari/assets/16781833/ee358bc8-9cb9-492c-a5e2-49465a6599f5)

In the end, I also think this logic is kind of a workaround to make the GUI look better, so for sure this is something to keep thinking about and improve!

Reply:
Agreed that in the long term the workaround should not be needed at all. I'm wary of adding more rendering state to the layer, but I'm hopeful we can find a good home for something like in the future if needed.